### ISSUE TITLE
ApolloClient exported by react-apollo isn't compatible with ApolloClient exported by apollo-client

### ISSUE THREAD

**Issue by jedwards1211** [flow-typed/flow-typed#2589]
@TLadd what do you think about just having `react-apollo` re-export types from `apollo-client`?

I am using `apollo-client`'s type in my `apollo-magic-refetch` package.  Could switch it to `react-apollo`'s type, but...it just seems wrong to redeclare all of this stuff in `react-apollo`

**Comment by jedwards1211** [comment]
Sigh, this shortcoming has been known since 2016 😞

**Comment by jedwards1211** [comment]
@TLadd I think we should just try to get the apollo folks to accept PRs to add `.js.flow` files to their packages, because that wouldn't suffer from this problem.  What do you think?

**Comment by TLadd** [comment]
Yeah, that would be the only way to manage them like this as is. I imagine it'll be an uphill battle to get that added back to the various apollo projects, since they saw the need to remove them once before. I'm not sure if it's really worth it to be honest. The react-apollo libdef gets you most of the functionality covered that is useful, at least in my experience. I recently moved my project over to typescript, so I'm probably not going to be much help going forward on maintaining this libdef.
